app/src/main/java/io/particle/android/sdk/ui/FlashAppHelper.java

```diff
@@ -63,22 +63,23 @@ static void flashKnownAppWithDialog(final FragmentActivity activity,
     // FIXME: remove duplication between the following methods
     private static void flashKnownApp(final Activity activity, final ParticleDevice device,
                                       final ParticleDevice.KnownApp knownApp) {
-        Async.executeAsync(device, new Async.ApiProcedure<ParticleDevice>() {
-            @Override
-            public Void callApi(@NonNull ParticleDevice sparkDevice) throws ParticleCloudException, IOException {
-                device.flashKnownApp(knownApp);
-                return null;
-            }
-
-            @Override
-            public void onFailure(@NonNull ParticleCloudException exception) {
-                new AlertDialog.Builder(activity)
-                        .setTitle("Unable to reflash " + capitalize(knownApp.getAppName()))
-                        .setMessage(exception.getBestMessage())
-                        .setPositiveButton(R.string.ok, (dialog, which) -> dialog.dismiss())
-                        .show();
-            }
-        }).andIgnoreCallbacksIfActivityIsFinishing(activity);
+        try {
```

Comment:
Wrapping every `Async` call in `try/catch` is pretty ugly.

Could `Async` be modified in a way that makes this automatic and doesn't incur any unwanted surprises on existing users?

Reply:
I think we were hitting task queue limit, there should be crashes related to AsyncTask in Fabric/Crashlytics. Did you get access to that?

Reply:
Just checked, yes it should catch RejectedExecutionException if it gets thrown.

Reply:
If it's the queue limit: it's 2018, and devices have far more resources to spare than when `Async` was started.  Let's just use our own ThreadPoolExecutor and give it a limit of, say, 256 tasks.  I don't want to push that behavior on everyone currently using the SDK though.  I'm thinking we could add an overload to allow specifying an executor in `executeAsync()`, or we could just add something like `Async.setExecutor()`.  (I'm leaning toward the latter.  Global, mutable state using static members on classes isn't great, but it's the least work for everyone by far, and it works.)

While I still think we should do the above, using our own executor with a greater capacity doesn't really fix the problem either. My thoughts on that:
1. We already check for `RejectedExecutionException` in one of the `executeAsync()` methods, we should probably do it in the other one, too.
2. We can't start throwing a new checked exception on the other method, since that would break API, but we could at least return a boolean indicating if the task was added to the queue. Again, returning a boolean isn't a great design, but it beats the current approach. :-/